Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs

```diff
@@ -26,80 +32,146 @@ public class GvrSceneManager : SingletonBehaviour<GvrSceneManager>
         /// </summary>
         public void LoadStartupScenes()
         {
-            generalScene = SceneManager.LoadScene(generalSceneName, new LoadSceneParameters(LoadSceneMode.Additive));
-            SceneManager.MoveGameObjectToScene(interactionManager, generalScene);
-
-            LoadWorld("world.zen", "FP_PICKRICE_SWAMP_02");
+            LoadWorld("oldmine.zen", "ENTRANCE_SURFACE_OLDMINE");
+            // PxCs.Interface.PxVm.CallFunction(GameData.I.VmGothicPtr, "STARTUP_SUB_OLDCAMP"); FP_GUARD_A_OC_179
         }
 
-        public void LoadWorld(string worldName, string startVob)
+        public async void LoadWorld(string worldName, string startVob)
         {
+            await ShowLoadingScene(worldName);
             newWorldName = worldName;
             startVobAfterLoading = startVob;
-            var newWorldScene = SceneManager.LoadScene(worldName, new LoadSceneParameters(LoadSceneMode.Additive));
+            var asyncLoad = SceneManager.LoadSceneAsync(worldName, new LoadSceneParameters(LoadSceneMode.Additive));
+            Scene newWorldScene;
 
             // Remove previous scene.
             // TODO - it might be, that we need to wait for old map to be removed before loading new one. Let's see...
             if (GameData.I.WorldScene.HasValue)
             {
+                // Try to delete everything in old scene, so we can load everything again from scratch
+                var objects = GameData.I.WorldScene.Value.GetRootGameObjects();
```

Comment:
We don't need to destroy objects from scene manually, do we? What happens if we just remove the scene? Everything should be gone. I don't want to iterate through the objects, please.

Reply:
This was more of a test that slipped up since as i mentioned on Discord, the gameobjects per scene are cached after being loaded once as so when we go to a world for a second time, instead of loading again the object from gothic files, we get them from memory in a blocking way, which i wanted to avoid for UX reasons (we want a loading screen between worlds, not to be stuck and load into a world without feedback on what's going on)